src/main/java/com/aws/greengrass/clientdevices/auth/configuration/RuntimeConfiguration.java

```diff
@@ -66,4 +90,87 @@ public void updateCAPassphrase(String passphrase) {
         caPassphrase.withValue(passphrase);
     }
 
+    /**
+     * Retrieve a Thing.
+     *
+     * @param thingName ThingName
+     * @return Optional of ThingV1 DTO, else empty optional
+     */
+    public Optional<ThingV1> getThingV1(String thingName) {
+        Topics v1ThingTopics = config.findTopics(THINGS_KEY, THINGS_V1_KEY, thingName);
+
+        if (v1ThingTopics == null) {
+            return Optional.empty();
+        }
+
+        Topics certTopics = v1ThingTopics.lookupTopics(THINGS_CERTIFICATES_KEY);
+        Map<String, Long> certMap = new HashMap<>();
+        certTopics.forEach(node -> {
+            certMap.put(node.getName(), Coerce.toLong(node));
+        });
+
+        return Optional.of(new ThingV1(thingName, certMap));
+    }
+
+    /**
+     * Store a Thing in the Runtime Configuration.
+     *
+     * @param thing Thing DTO
+     */
+    public void putThing(ThingV1 thing) {
+        if (thing == null) {
+            return;
+        }
+
+        Topics v1ThingTopics = config.lookupTopics(THINGS_KEY, THINGS_V1_KEY, thing.getThingName());
+        Map<String, Object> certMap = new HashMap<>(thing.getCertificates());
+        v1ThingTopics.lookupTopics(THINGS_CERTIFICATES_KEY).replaceAndWait(certMap);
+    }
+
+    /**
+     * Get a certificate.
+     *
+     * @param certificateId Certificate ID
+     * @return  Optional of CertificateV1 DTO, else empty optional
+     */
+    public Optional<CertificateV1> getCertificateV1(String certificateId) {
+        Topics v1CertTopics = config.findTopics(CERTS_KEY, CERTS_V1_KEY, certificateId);
+
+        if (v1CertTopics == null) {
+            return Optional.empty();
+        }
+
+        Topic statusTopic = v1CertTopics.find(CERTS_STATUS_KEY);
+        Topic statusUpdatedTopic = v1CertTopics.find(CERTS_STATUS_UPDATED_KEY);
+
+        CertificateV1.Status status = CertificateV1.Status.UNKNOWN;
+        if (statusTopic != null) {
+            int topicVal = Coerce.toInt(statusTopic);
+            if (topicVal < CertificateV1.Status.values().length) {
+                status = CertificateV1.Status.values()[topicVal];
+            }
+        }
+
+        Long statusUpdated = 0L;
+        if (statusUpdated != null) {
+            statusUpdated = Coerce.toLong(statusUpdatedTopic);
+        }
+
+        return Optional.of(new CertificateV1(certificateId, status, statusUpdated));
+    }
+
+    /**
+     * Store a certificate in Runtime Configuration.
+     *
+     * @param cert Certificate DTO
+     */
+    public void putCertificate(CertificateV1 cert) {
+        if (cert == null) {
```

Comment:
Should we throw an error? Or at least log something, just doing this might lead to someone thinking it did store somethings (not supper critical though). Same comment for the other place we check for `null`